src/core/library.c

```diff
@@ -1112,6 +1112,31 @@ QuicLibrarySetGlobalParam(
         Status = QUIC_STATUS_SUCCESS;
         break;
 
+    case QUIC_PARAM_GLOBAL_STATELESS_RESET_KEY:
+        if (BufferLength != QUIC_STATELESS_RESET_KEY_LENGTH * sizeof (uint8_t)) {
+            Status = QUIC_STATUS_INVALID_PARAMETER;
+            break;
+        }
+
+        CXPLAT_HASH* tokenHash = NULL;
+        Status =
+            CxPlatHashCreate(
+                CXPLAT_HASH_SHA256,
+                (uint8_t*)Buffer,
+                QUIC_STATELESS_RESET_KEY_LENGTH * sizeof(uint8_t),
+                &tokenHash);
```

Comment:
You still need to do this in the loop, once per proc.

Reply:
In this case, if CxPlatHashCreate failed in the middle of the loop, should we free all proc's ResetTokenHash?

Reply:
Code like below looks right?

    case QUIC_PARAM_GLOBAL_STATELESS_RESET_KEY:
        if (BufferLength != QUIC_STATELESS_RESET_KEY_LENGTH * sizeof (uint8_t)) {
            Status = QUIC_STATUS_INVALID_PARAMETER;
            break;
        }

		Status = QUIC_STATUS_SUCCESS;
        for (uint16_t i = 0; i < MsQuicLib.ProcessorCount; ++i) {
            CXPLAT_HASH* tokenHash = NULL;
            Status =
                CxPlatHashCreate(
                    CXPLAT_HASH_SHA256,
                    (uint8_t*)Buffer,
                    QUIC_STATELESS_RESET_KEY_LENGTH * sizeof(uint8_t),
                    &tokenHash);
            if (QUIC_SUCCEEDED(Status))
            {
                QUIC_LIBRARY_PP* PerProc = &MsQuicLib.PerProc[i];
                CxPlatLockAcquire(&PerProc->ResetTokenLock);
                CxPlatHashFree(PerProc->ResetTokenHash);
                PerProc->ResetTokenHash = tokenHash;
                CxPlatLockRelease(&PerProc->ResetTokenLock);
            } else {				
				for (uint16_t i = 0; i < MsQuicLib.ProcessorCount; ++i) {
					// should we free all tokenHash?
					QUIC_LIBRARY_PP* PerProc = &MsQuicLib.PerProc[i];
					CxPlatLockAcquire(&PerProc->ResetTokenLock);
					CxPlatHashFree(PerProc->ResetTokenHash);
					PerProc->ResetTokenHash = NULL;
					CxPlatLockRelease(&PerProc->ResetTokenLock);					
				}
				break;
            }
        }       
        break;

Reply:
No need to have the other loop that cleans up. If you fail to create a new hash, you just leave the old/other ones and bail.